tests/ui_tests/app/page/cluster_list/page.py

```diff
@@ -193,6 +200,11 @@ def run_upgrade_in_cluster_row(
             if self.is_element_displayed(ActionDialog.next_btn):
                 self.find_and_click(ActionDialog.next_btn)
         if hc_acl:
+            if is_new_host:
+                cluster_components_page = ClusterComponentsPage(self.driver, self.base_url, 1)
+                cluster_components_page.click_add_host_btn()
+                cluster_components_page.host_popup.create_host("Test_host")
+                self.wait_element_hide(HostCreationLocators.block)
```

Comment:
It's very confusing why we use another page to run something.

Maybe we may move `click_add_host_btn` to some generic class/function that can take driver and maybe some info on which locator to use (like `in_context: Literal["popup", "cluster-components", "host-list", ...]`) and inside the function there'll be routing based on case.

Maybe it should be like `HostCreator` or something like that, so it'll have public methods for each case (e.g. `on_host_list_page`, `on_upgrade_popup`, `on_cluster_components_page`) and fewer private implementation functions.

What do you think?

Reply:
yep, that's good idea. moved to special class everything about host-component